### PR TITLE
fix `reserialize()` of `InternalIntlMessage`

### PR DESCRIPTION
**Pull request by aamiaa**
This fixes `serializeAst()` and `serializeAstTag()` used by `InternalIntlMessage.reserialize()`.

I've tested the function on intl messages used by the Discord client and compared with their versions from the old system. Most of them are the exact same, with the rest being only slightly different (aka it works well!).

The only thing I'm unsure about is the node index in `$link` handler. The only enum values with index 3 are `Offset` and `Control`, none of which seem to fit here so I left it hardcoded.

### CODE REVIEW

File: packages/intl/src/message.ts [hunk 1]
```diff
@@ -103,38 +105,38 @@ function serializeAstTag(node: TagNode, result: { value: string }) {
...
     case '$p':
       serializeAst(node[AstNodeIndices.Children], result);
       result.value += '\n\n';
-      return;
+      break;
     case '$link':
       // The target is the first child of the link. We don't have to care if it's a placeholder
       // or not, because the serialization will automatically remove the extra empty.
-      const [target, ...children] = node[AstNodeIndices.Children];
+      const children = node[AstNodeIndices.Children], target = node[3]
```
Comment (faultyserver) [inline packages/intl/src/message.ts:126]: `Control` is actually the right one here! it's not super obvious, but here's the [definition context](https://github.com/discord/discord-intl/blob/1cab2d38e0581e166e4709de4958307902c04edc/crates/intl_markdown/src/icu/compile.rs#L69-L79), [type reference for consumers](https://github.com/discord/discord-intl/blob/1cab2d38e0581e166e4709de4958307902c04edc/packages/intl/src/types.d.ts#L195), and [usage in the formatter](https://github.com/discord/discord-intl/blob/1cab2d38e0581e166e4709de4958307902c04edc/packages/intl/src/format.ts#L148-L158).

tldr; Links are `Tag` elements, and tags have the additional `control` field to hold non-visible information, in this case the link target.

```suggestion
      const children = node[AstNodeIndices.Children];
      const target = node[AstNodeIndices.Control];
```

**Review by faultyserver — COMMENTED** [review]
Thanks for fixing this! Eventually we should add some tests for the runtime to make sure these things don't regress again, but they don't exist yet.

One small note about the `Control`, but LGTM
